Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs

```diff
@@ -349,6 +331,49 @@ private static bool IsNonRetriableException(Exception exception)
                 return false;
             }
 
+            /// <summary>
+            /// Returns an enumerator containing the private and regional service endpoints to iterate over.
+            /// </summary>
+            /// <param name="defaultEndpoint">An instance of <see cref="Uri"/> containing the default global endpoint.</param>
+            /// <param name="locations">An instance of <see cref="IList{T}"/> containing the preferred location names.</param>
+            /// <param name="accountInitializationCustomEndpoints">An instance of <see cref="IList{T}"/> containing the custom private endpoints.</param>
+            /// <returns>An instance of <see cref="IEnumerator{T}"/> containing the service endpoints.</returns>
+            private static IEnumerator<Uri> GetServiceEndpointEnumerator(
+                Uri defaultEndpoint,
+                IList<string>? locations,
+                IList<string> accountInitializationCustomEndpoints)

```

Comment:
If we compare this with `null`, then should this be 
```suggestion
                IList<string>? accountInitializationCustomEndpoints)

```

And use instead `HasValue`?

Reply:
Looks like, even the nullable collections `<IList<string>? list` doesn't have a definition of `HasValue`. 

![image](https://github.com/Azure/azure-cosmos-dotnet-v3/assets/87335885/3694616e-3504-40b7-9af2-879fa54060a3)